### PR TITLE
test caching of DNSSEC validation and of DNSSEC records

### PR DESCRIPTION
**Pull request by japaric**
hickory was failing the 'section 4.5' test before #2239 was merged but now it passes it

### CODE REVIEW

**Comment by bluejekyll** [comment]
I think this is waiting for a review from your end @japaric, let me know when ready and I can merge.

File: conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_4/section_4_5.rs
Comment (justahero) [inline conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_4/section_4_5.rs:43]: Feel free to ignore, it seems the `match` statement expresses a concept.

```rust
impl Direction {
    pub fn peer(&self) -> &Ipv4Addr {
        match self {
            Direction::Incoming { source } => source,
            Direction::Outgoing { destination } => destination,
        }
    }
}
```

**Review by justahero — APPROVED** [review]
Left a small suggestion, otherwise looks good.

File: conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_4/section_4_5.rs
Comment (japaric) [reply conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_4/section_4_5.rs:43]: good idea. added

**Comment by japaric** [comment]
rebased. I had to un-ignored one test since it passes now that #2253 has landed.

**Review by pvdrz — APPROVED** [review]

**Pull request merged**
